Sources/HummingbirdCore/Response/ResponseBody.swift

```diff
@@ -50,6 +53,30 @@ public struct HBResponseBody: Sendable {
             for try await buffer in asyncSequence {
                 try await writer.write(buffer)
             }
+            return
         }
     }
+
+    /// Create HBResponseBody that returns trailing headers from its closure once all the 
+    /// body parts have been written 
+    /// - Parameters:
+    ///   - contentLength: Optional length of body
+    ///   - write: closure provided with `writer` type that can be used to write to response body
+    ///         trailing headers are returned from the closure after all the body parts have been
+    ///         written
+    public static func withTrailingHeaders(
+        contentLength: Int? = nil, 
+        _ write: @Sendable @escaping (any HBResponseBodyWriter) async throws -> HTTPFields?
```

Comment:
I like this a lot better, now one more idea to play with: Can we make the `any HBResponseBodyWriter` an `inout`? This would make it harder to incorrectly wield this API.

Reply:
I'm not sure what you are getting at there. What do we gain from  making `any HBResponseBodyWriter` `inout`. That closure is called by `HTTPChannelHandler` and any Middleware that want to process response bodies. ie very few people will be calling it.

Are you seeing a situation where you would want to the write function to mutate the `HBResponseBodyWriter`?

Reply:
No, I don't see a situation where we'd want to mutate. I see a risk in people trying to escape that HBResponseBodyWriter out of this closure

Reply:
I don't want to hold up this PR for this topic though, since it's not caused by this PR

Reply:
ok I see, by making it inout you make it harder but it could still be done. I could make `~Copyable` I don't know if that'd help

Reply:
Forgot that HBResponseBodyWriter is a protocol not a struct

Reply:
Adding an issue to chase this up